modules/mod_ls.c

```diff
@@ -1234,16 +1234,21 @@ static char **sreaddir(const char *dirname, const int sort) {
 
   pr_fs_clear_cache2(dirname);
   if (pr_fsio_stat(dirname, &st) < 0) {
+    pr_log_debug(DEBUG8, "error checking '%s': %s",
+      dirname, strerror(errno));
```

Comment:
FYI, you may notice in some places in the code that I use a pattern like this:
```
if (func(...) < 0) {
  int xerrno = errno;

  pr_log_debug(..., "some error occurred: %s", strerror(xerrno));
  errno = xerrno;
  return -1;
}
```

I often use a temporary `xerrno` variable to "cache" the actual `errno` value returned by the function, so that I can properly set/return that `errno` value for _this_ function's caller.  In the past, functions like `pr_log_debug()` (and others) have changed the global `errno` values themselves -- and that means that without careful handling, the caller might see the `errno` value as set by `pr_log_debug()`, and not by the actual function.

Reply:
Thanks for the info, good to know that. I will change the pattern for all places.